lib/backend/k8s/resources/customresource.go

```diff
@@ -310,3 +326,63 @@ func (c *customK8sResourceClient) Watch(ctx context.Context, list model.ListInte
 func (c *customK8sResourceClient) EnsureInitialized() error {
 	return nil
 }
+
+func listInterfaceToKey(l model.ListInterface) model.Key {
+	pl := l.(model.ResourceListOptions)
+	if pl.Name != "" {
+		return model.ResourceKey{Name: pl.Name, Kind: pl.Kind}
+	}
+	return nil
+}
+
+func keyToName(k model.Key) (string, error) {
+	return k.(model.ResourceKey).Name, nil
+}
+
+func convertResourceToKVPair(r Resource, resourceKind string) (*model.KVPair, error) {
+	kvp := &model.KVPair{
+		Key: model.ResourceKey{
+			Name:      r.GetObjectMeta().GetName(),
+			Namespace: r.GetObjectMeta().GetNamespace(),
+			Kind:      resourceKind,
+		},
+		Revision: r.GetObjectMeta().GetResourceVersion(),
+	}
+
+	var v2resource Resource
```

Comment:
This isn't necessary.  The resource that is passed in is already the correct type.